prow/plugins/trigger/pr.go

```diff
@@ -65,6 +65,10 @@ func handlePR(c client, trustedOrg string, pr github.PullRequestEvent) error {
 			if err != nil {
 				c.Logger.Warnf("Failed to clear stale comments: %v.", err)
 			}
+			// Just try to remove "needs-ok-to-test" label if existing, we don't care about the result.
+			if err = c.GitHubClient.RemoveLabel(trustedOrg, pr.PullRequest.Base.Repo.Name, pr.PullRequest.Number, needsOkToTest); err != nil {
+				c.Logger.Warnf("Failed at removing %s label: %v", needsOkToTest, err)
```

Comment:
Will this error every time the label does not exist?

Reply:
I'm not sure about that. I guess the server will throw a 404 and the client take it as an error. However, I don't want to list all labels first and check if `needs-ok-to-test` exists. (this will increase an API token). How about ignore the result of `RemoveLabel` here?

@cjwagner WDYT?

Reply:
Just ignore the error.

I'm concerned about token usage from recent changes to this code, specifically the change that makes PR events always list all the comments whereas previously comments were only listed for non-member PRs. After this change, every time an org-member updates a PR (a frequent event) we will list comments and try to remove the `needs-ok-to-test` label which requires at least two additional tokens. All of this is necessary to catch the rare case in which a PR starts as untrusted, but becomes trusted when the PR author becomes an org member.

I propose that we only handle this case for PR reopen events and ignore it for synchronize event. The reopen event is rare compared to the synchronize event so this would save a lot of tokens, but if a PR author starts as a non-member and becomes a member, they can make their PR trusted by reopening it.
WDYT?

Reply:
Agreed.

>but if a PR author starts as a non-member and becomes a member, they can make their PR trusted by reopening it.

This makes sense. Another approach might be that the author give an "ok-to-test" comment for himself/herself. I guess it also works, right?